### PR TITLE
Add (de)serialization for Property/Records

### PR DESCRIPTION
**Pull request by M3rs**
Properties have a backing field named like '<MyField>k__BackingField'
which can get used for properties and record types.

i.e.

```
public class MyClass
{
	public string MyString { get; set; }
}
```

When looking at `typeof(MyClass)`, there will be a private field
`private string <MyString>k__BackingField`.

- Add to `GetCompositeDeserializer` and `GetCompositeSerializer` the
  ability to read properties.
- Add to tests examples of class with Properties and record type

### CODE REVIEW

**Review by SamboyCoding — CHANGES_REQUESTED** [review]
Please gitignore your .vs folder - test logs and user-specific configuration should not be pushed.

File: Tomlet/TomlSerializationMethods.cs [hunk 1]
```diff
@@ -136,7 +136,7 @@ internal static Deserialize<object> GetCompositeDeserializer(Type type)
             else
             {
                 //Get all instance fields
-                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.SetProperty);
```
Comment (SamboyCoding) [inline Tomlet/TomlSerializationMethods.cs]: I don't think this change is required, `SetProperty` only applies when calling methods, from what I can see on the docs, and regardless your change isn't required by the code you wrote below.

File: Tomlet/TomlSerializationMethods.cs [hunk 2]
```diff
@@ -222,7 +227,7 @@ internal static Serialize<object> GetCompositeSerializer(Type type)
             {
 
                 //Get all instance fields
-                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.GetProperty);
```
Comment (SamboyCoding) [inline Tomlet/TomlSerializationMethods.cs]: Same as above, I don't think `GetProperty` needs to be specified here.

**Review by SamboyCoding — CHANGES_REQUESTED** [review]

**Comment by SamboyCoding** [comment]
Other than those minor details, nice PR! Very appreciated, and thanks for writing additional tests too.

**Comment by M3rs** [comment]
Ah, good call out on the BindingFlags. Cleaned up that and the gitignore.

**Review by SamboyCoding — APPROVED** [review]

**Comment by SamboyCoding** [comment]
Alright, merging this now, will try and get a release out tonight.

**Comment by SamboyCoding** [comment]
As a slight aside, your local git email doesn't match your github one, so these commits aren't attributed to you.

**Comment by M3rs** [comment]
Ah, I'll get that sorted, thanks

**Comment by SamboyCoding** [comment]
This is now live in 2.0.0. Sorry for the delay in getting this out in a release!

**Pull request merged**